pkgs/http_client_conformance_tests/lib/src/redirect_tests.dart

```diff
@@ -2,60 +2,40 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'dart:async';
-import 'dart:io';
-
 import 'package:http/http.dart';
+import 'package:stream_channel/stream_channel.dart';
 import 'package:test/test.dart';
 
 /// Tests that the [Client] correctly implements HTTP redirect logic.
 void testRedirect(Client client) async {
   group('redirects', () {
-    late HttpServer server;
+    late String host;
+    late StreamChannel<Object?> httpServerChannel;
+
     setUp(() async {
```

Comment:
Should we consider using `setUpAll`? The server looks stateless so we shouldn't lose much safety, but we can save some time by using a single isolate and server.

Reply:
The server is stateless but the guidance for `setUpAll` is to avoid it unless `setUp` is prohibitively slow (which it isn't). 

But I don't have a strong opinion here.